HotelBooking.API/Controllers/HotelController.cs

```diff
@@ -87,9 +87,102 @@ public async Task<IActionResult> CreateFacilityAsync([FromForm] FacilityRequest
             });
         }
 
+        [Authorize(Roles = "Administrator", AuthenticationSchemes = "Bearer")]
+        [HttpPut("update-facility")]
+        public async Task<IActionResult> UpdateFacilityAsync(FacilityModel model, Guid id)
+        {
+            if (model == null)
+            {
+                logger.LogError("Invalid model sent from client.");
+                return BadRequest("Invalid model object");
+            }
+            var result = await hotelService.UpdateFacilityAsync(model, id);
+            if (result)
+            {
+                return StatusCode(StatusCodes.Status200OK, new ResponseModel
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    IsSuccess = true,
+                    Data = new
+                    {
+                        Guid = id
+                    }
+                });
+            }
+            return StatusCode(StatusCodes.Status400BadRequest, new ResponseModel
+            {
+                StatusCode = HttpStatusCode.BadRequest,
+                IsSuccess = false
+            });
+        }
+
+        [Authorize(Roles = "Administrator", AuthenticationSchemes = "Bearer")]
+        [HttpDelete("delete-facility")]
+        public async Task<IActionResult> DeleteFacility(Guid id)
+        {
+            var result = await hotelService.DeleteFacilityAsync(id);
+            if (result)
+            {
+                return StatusCode(StatusCodes.Status200OK, new ResponseModel
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    IsSuccess = true,
+                    Data = new
+                    {
+                        Guid = id
+                    }
+                });
+            }
+            return StatusCode(StatusCodes.Status400BadRequest, new ResponseModel
+            {
+                StatusCode = HttpStatusCode.BadRequest,
+                IsSuccess = false
+            });
+        }
+
+        [HttpGet("get-all-facilities")]
+        public async Task<IActionResult> GetAllFacilities()
+        {
+            var result = await hotelService.GetAllFacilities();
+            if (result != null)
+            {
+                return StatusCode(StatusCodes.Status200OK, new ResponseModel
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    IsSuccess = true,
+                    Data = result
+                });
+            }
+            return StatusCode(StatusCodes.Status404NotFound, new ResponseModel
+            {
```

Comment:
you can write it in a better way
return result != null ? Ok() : NotFound()
don't need if else for this case

Reply:
I fixed it. thanks.